### ISSUE TITLE
For the purposes of pdfsizeopt, should this be preferred instead of sam2p?

### ISSUE THREAD

**Issue by rbrito** [pts/imgdataopt#3]
Hi, @pts.

I am packaging pdfsizeopt and I have something that works, but, unfortunately, if I use the usual Debian toolchain with calling autoreconf on sam2p, I get the compilation to bomb (it only works if I call `./configure` directly).

OTOH, imgdataopt seems much simpler and smaller. Would I be loosing anything if I used imgdataopt instead of sam2p?

For some kind of images (sometimes, small images, but I can't pinpoint exactly which ones), sam2p (perhaps sam2p_np, but I may be mistaken) wins over all other much more advanced image optimizers (even the very slow ones, like optipng+advdef, pngout, zopflipng and so on). Sometimes, even by quite a few percent.

Would I be losing such benefits of sam2p if I used imgdataopt instead?

Thanks,

Rogério Brito.

**Comment by pts** [comment]
**TL;DR** Yes.

imgdataopt is a rewrite of sam2p in C with the same features (as far as pdfsizeopt is concerned) but fewer dependencies. See https://github.com/pts/pdfsizeopt/issues/51 for more comments on the motivation and initial design of imgdataopt.

As of now, when called from pdfsizeopt, sam2p and imgdataopt should create bit-by-bit identical output (except when imgdataopt creates a PNG, and sam2p creates a PDF, but the underlying image data must be identical), and pdfsizeopt should create an identical output PDF. If there is a difference, it should be treated as a bug, and fixed in imgdataopt (or, maybe, sam2p).

> Would I be loosing anything if I used imgdataopt instead of sam2p?
> Would I be losing such benefits of sam2p if I used imgdataopt instead?

No, imgdataopt was created to copy all features of sam2p needed by pdfsizeopt.
